externs/firebase-app-externs.js

```diff
@@ -273,3 +273,38 @@ firebase.Promise.reject = function(error) {};
  * @return {!firebase.Promise<!Array<*>>}
  */
 firebase.Promise.all = function(values) {};
+
+/** @interface */
+firebase.Observer = function() {};
+
+/**
+ * @template V
+ * @param {?V} value
+ */
+firebase.Observer.prototype.next = function(value) {};
+
+/**
+ * @template E
+ * @param {!E} error
+ */
+firebase.Observer.prototype.error = function(error) {};
+
+firebase.Observer.prototype.complete = function() {};
+
+/**
```

Comment:
Unfortunately Closure doesn't support templated typedefs:

```
../../externs/firebase-app-externs.js:296: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @typedef {function(?V): void}
   ^

../../externs/firebase-app-externs.js:298: WARNING - Misplaced template annotation. @template is only allowed in class, constructor, interface, function or method declarations
firebase.NextFn;
^^^^^^^^^^^^^^^

../../externs/firebase-app-externs.js:302: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @typedef {function(!E): void}
   ^

../../externs/firebase-app-externs.js:304: WARNING - Misplaced template annotation. @template is only allowed in class, constructor, interface, function or method declarations
firebase.ErrorFn;
^^^^^^^^^^^^^^^^
```

It's probably best to leave the NextFn and ErrorFn arguments as raw "!function(...)" annotations instead of adding the typedefs here. I'd be fine if you left CompleteFn and Unsubscribe in as typedefs and added a comment about the templated typedefs being unfeasible or you just removed all of them. Either one is good with me.